alf/algorithms/sac_algorithm.py

```diff
@@ -226,10 +226,10 @@ def _actor_train_step(self, exp: Experience, state: SacActorState,
 
             target_q_value = torch.min(critic1, critic2)
 
-            dqda = torch.autograd.grad(target_q_value, action,
-                                       torch.ones(target_q_value.size()))[0]
-
-            def actor_loss_fn(dqda, action):
+            def actor_loss_fn(action):
```

Comment:
It's better to do grad only once if possible. The following might work:
```python
        dqda = alf.nest.pack_sequence_as(action,
            torch.autograd.grad(target_q_value, alf.nest.flatten(action)))
```

Reply:
Now changed to one pass computation as suggested.